### PR TITLE
Java Docs updates and fixes

### PR DESCRIPTION
**Pull request by aarohib**
## What does this PR do?
Catch-all PR for typos, updating h4s, breaking down content for the App Dev Guide Foundations + Features.

### CODE REVIEW

**Review by flossypurse — APPROVED** [review]
run `yarn gen` to update the guides as well.

**Comment by aarohib** [comment]
Yeah, still working through some other changes that I'll be pushing soon. Will run it once I have them all in before opening this up for review.

File: docs/go/how-to-customize-activity-type-in-go.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+id: how-to-customize-activity-type-in-go
+title: How to customize Activity Type in Go
+sidebar_label: Customize Activity Type
+description: To customize the Workflow Type set the `Name` parameter with `RegisterOptions` when registering your Workflow with a Worker.
+tags:
+  - developer-guide
+  - go
+---
+
+To customize the Activity Type set the `Name` parameter with `RegisterOptions` when registering your Activity with a Worker.
```
Comment (djmagee) [inline docs/go/how-to-customize-activity-type-in-go.md:11]:
```suggestion
To customize the Activity Type, set the `Name` parameter with `RegisterOptions` when registering your Activity with a Worker.
```

File: docs/go/how-to-customize-workflow-type-in-go.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+id: how-to-customize-workflow-type-in-go
+title: How to customize Workflow Type in Go
+sidebar_label: Customize Workflow Type
+description: To customize the Workflow Type set the `Name` parameter with `RegisterOptions` when registering your Workflow with a Worker.
+tags:
+  - developer-guide
+  - go
+---
+
+To customize the Workflow Type set the `Name` parameter with `RegisterOptions` when registering your Workflow with a Worker.
```
Comment (djmagee) [inline docs/go/how-to-customize-workflow-type-in-go.md:11]:
```suggestion
To customize the Workflow Type, set the `Name` parameter with `RegisterOptions` when registering your Workflow with a Worker.
```

**Review by djmagee — APPROVED** [review]

**Pull request merged**